protocol/backend/backend.go

```diff
@@ -2,82 +2,51 @@ package backend
 
 import (
 	"context"
-	"crypto/ecdsa"
-	"math/big"
 	"sync"
 	"time"
 
-	eth "github.com/ethereum/go-ethereum"
-	"github.com/ethereum/go-ethereum/accounts/abi/bind"
 	ethcommon "github.com/ethereum/go-ethereum/common"
-	ethtypes "github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/ethclient"
 
 	"github.com/athanorlabs/atomic-swap/common"
 	"github.com/athanorlabs/atomic-swap/common/types"
 	mcrypto "github.com/athanorlabs/atomic-swap/crypto/monero"
 	contracts "github.com/athanorlabs/atomic-swap/ethereum"
-	"github.com/athanorlabs/atomic-swap/ethereum/block"
+	"github.com/athanorlabs/atomic-swap/ethereum/extethclient"
 	"github.com/athanorlabs/atomic-swap/monero"
 	"github.com/athanorlabs/atomic-swap/net"
 	"github.com/athanorlabs/atomic-swap/protocol/swap"
 	"github.com/athanorlabs/atomic-swap/protocol/txsender"
-
-	logging "github.com/ipfs/go-log"
 )
 
 var (
-	log                    = logging.Logger("protocol/backend")
 	defaultTimeoutDuration = time.Hour * 24
 )
 
 // Backend provides an interface for both the XMRTaker and XMRMaker into the Monero/Ethereum chains.
 // It also interfaces with the network layer.
 type Backend interface {
-	monero.WalletClient
+	XMR() monero.WalletClient
+	ETH() extethclient.EthClient
```

Comment:
```suggestion
	XMRClient() monero.WalletClient
	ETHClient() extethclient.EthClient
```

nit but I'd rather name them something like this? otherwise it's a bit unclear imo

Reply:
It was one of the things I wanted to discuss with you. I originally went with what you were suggesting, but they are used in so many places that brevity might be a bigger code-readability win. It's a compromise either way. I can create another branch so you can see the difference side-by-side.